src/coprocessor/dag/expr/builtin_math.rs

```diff
@@ -878,15 +879,35 @@ mod tests {
     #[test]
     fn test_rand_with_seed() {
         let seed: i64 = 20160101;
+        let expect = eval_func(ScalarFuncSig::RandWithSeed, &[Datum::I64(seed)])
+            .unwrap()
+            .as_real()
+            .unwrap()
+            .unwrap()
+            .to_bits();
         for _ in 1..3 {
             let got = eval_func(ScalarFuncSig::RandWithSeed, &[Datum::I64(seed)])
                 .unwrap()
                 .as_real()
                 .unwrap();
 
             assert!(got.is_some());
-            assert_eq!(got.unwrap().to_bits(), (0.4545469470152683f64).to_bits());
+            assert_eq!(got.unwrap().to_bits(), expect);
+        }
+        // TODO: this `expect_same_probability` need to be re-selected carefully
+        let expect_same_probability = 0.0001;
```

Comment:
This probability is both useless and too lenient. 

The test evaluates 1024 numbers seeded with various data and checks if there are any duplicates. As soon as there is a single duplicate, the ratio (1023/1024) ≈ **0.9990** is already less than (1 - 0.0001) = **0.9999**, and thus the `assert!(x/y > 1-p)` is equivalent to `assert!(x == y)` anyway.

And then it is too lenient. If the random numbers generated are evenly distributed, the collision rate should be 2<sup>-53</sup>. With 1024 numbers, the probability at *any two* of those 1024 elements be the same [would be](https://en.wikipedia.org/wiki/Birthday_problem#Approximations)

      1 - exp(-1024^2 / (2*2^53))
    = 5.8e-11

and it will take `test_cnt >= 130000` before reaching a collision probability of 0.0001. 

Finally, since the PRNG algorithm is deterministic, if a collision does actually occur, we need to modify the test case or the algorithm anyway, using a probabilistic measure here is pointless.

So I'd suggest:

* Remove the `expect_same_probability`,
* Change the `assert!` to `assert_eq!(set.len(), test_cnt)`
* If, in the future, a collision does happen but not due to logical error, we just choose a different `seed` to start.

Reply:
Ok, I will fix it